src/crimson/os/seastore/segment_cleaner.h

```diff
@@ -37,6 +37,7 @@ class SegmentCleaner : public JournalSegmentProvider {
   /// Callback interface for querying and operating on segments
   class ExtentCallbackInterface {
   public:
+    virtual ~ExtentCallbackInterface() = 0;
```

Comment:
There is an implementation of that one, right?
(required for pure virtual destructors)

Reply:
i don't follow you. what do you mean by "that one"? i could instead put
```c++
virtual ~ExtentCallbackInterface() = default;
```
or
```c++
virtual ~ExtentCallbackInterface() {}
```
but since this is an abstract class. i think it'd be better to make this explicit.

Reply:
Take a look at https://godbolt.org/z/jvjzsK
or
"A prospective (since C++20)destructor may be declared pure virtual, for example in a base class which needs to be made abstract, but has no other suitable functions that could be declared pure virtual. **A pure virtual destructor must have a definition,** since all base class destructors are always called when the derived class is destroyed"

Reply:
@ronen-fr thanks for pointing this out. fixed and repushed.